### ISSUE TITLE
GCC complains about thing might be uninitialized

### ISSUE THREAD

**Issue by albinahlback** [wbhart/flint2#1080]
Ping @tthsqe12. I don't know how to suppress this warning. If I understand it correctly, it will never get uninitialized, but because `N` is a slong (see the code for its use), the compiler cannot prove that it will get initialized.
```
min_fields.c: In function ‘mpoly_min_fields_ui_sp’:
min_fields.c:44:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
   44 |     mpoly_unpack_vec_ui(min_fields, pmin, bits, mctx->nfields, 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from min_fields.c:12:
/home/albinah/cpkg/flint2/mpoly.h:1108:16: note: by argument 2 of type ‘const mp_limb_t *’ {aka ‘const long unsigned int *’} to ‘mpoly_unpack_vec_ui’ declared here
 1108 | FLINT_DLL void mpoly_unpack_vec_ui(ulong * exp1, const ulong * exp2,
      |                ^~~~~~~~~~~~~~~~~~~
min_fields.c: In function ‘mpoly_min_fields_fmpz’:
min_fields.c:83:5: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
   83 |     mpoly_unpack_vec_fmpz(min_fields, pmin, bits, mctx->nfields, 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from min_fields.c:12:
/home/albinah/cpkg/flint2/mpoly.h:1111:16: note: by argument 2 of type ‘const mp_limb_t *’ {aka ‘const long unsigned int *’} to ‘mpoly_unpack_vec_fmpz’ declared here
 1111 | FLINT_DLL void mpoly_unpack_vec_fmpz(fmpz * exp1, const ulong * exp2,
      |                ^~~~~~~~~~~~~~~~~~~~~
```

**Comment by albinahlback** [comment]
I suggest using some pragmas in these cases. I believe there are two within the whole repository.

**Comment by tthsqe12** [comment]
I there are other places as well. I will see what I can do.

**Comment by albinahlback** [comment]
My PR will handle this by using pragmas.

**Comment by tthsqe12** [comment]
How is this warning message related to N being an slong?

**Comment by tthsqe12** [comment]
related list of bugs:
https://gcc.gnu.org/bugzilla/buglist.cgi?quicksearch=may%20be%20uninitialized

**Comment by wbhart** [comment]
As this is a warning only and @tthsqe12 and I feel the code is 100% valid (so that the warning is invalid), I suggest we hold over this ticket until the next release.

**Comment by wbhart** [comment]
@albinahlback Are you able to explain why you believed that N is an slong is the reason for this issue? @tthsqe12 and I weren't able to see why that should be a problem.

**Comment by albinahlback** [comment]
Yeah, I do not get why I said that. Moreover, I do not get that error anymore with GCC 12.
